### ISSUE TITLE
the question about BPE

### ISSUE THREAD

**Issue by travel-go** [pytorch/fairseq#150]
hello,I want to bpe the en-de data.I know that both ends of the training set need to do BPE and the test set also needs to do BPE, ut I'm not sure if the valid set's English do BPE, what about German?
Is it required to do BPE in English and German for valid set?
If you have spare time,please give me an answer.

**Comment by michaelauli** [comment]
The short answer is, yes the valid set also needs to be tokenized with BPE. Check the script here on how we do BPE tokenization: https://github.com/pytorch/fairseq/blob/master/data/prepare-wmt14en2de.sh

**Comment by travel-go** [comment]
Thank you for your reply.Trouble you a lot,really thank you very much.

**Comment by MikhailSam** [comment]
@michaelauli 
HI! Can you please update the link -  looks alike it is broken now :(
I'm really interested in it too!
